src/read_params.jl

```diff
@@ -0,0 +1,45 @@
+
+# ------------------------------------------
+#   Read ACE fit parameters from file 
+
+using ACE1pack
+
```

Comment:
Following discussion in #3, one actually doesn't need to use any of JuLIP dictionary reading here, that should be done by the user, e.g. 

```
using JuLIP
fit_params = load_dict(@__DIR__() * "/files/fit_params.json")
```
Since that's a dictionary already and dictionaries, not composite types are used throughout ACE1pack, there's no need to define a `read_dict` here. 

So what's left is a `fill_defaults!` utility function, which I think acts differently enough from `read_dict` to be named distinctly. 

I think the most common way of loading the parameters ahead of fitting will be
```
using ACE1pack, JuLIP
fit_params = fill_defaults!(load_dict(@__DIR__() * "/files/fit_params.json"))
fit_ace(fit_params)
```

Reply:
Hm ... are you saying you want to read a dictionary with partial parameters and then fill in the defaults? Ok, I should have anticipated the need for this. Let me sit on this for a moment please.

Reply:
Ok, I think I know how to do some of this: 
```Julia
# this function just shows the kwargs so you can see how the code works.
f(; kwargs...) = (@show kwargs)
```
e.g., 
```
julia> f(;a = 1, b = 2)
kwargs = Base.Pairs(:a => 1, :b => 2)
pairs(::NamedTuple) with 2 entries:
  :a => 1
  :b => 2
```

In principle you can splat a dictionary using `D...` to turn it into kwargs, but the problem is that the keys are strings and not symbols. E.g., 
```Julia 
julia> D = Dict("a" => 1, "b" => 2); f(; D...)
ERROR: TypeError: in typeassert, expected Symbol, got a value of type String
Stacktrace:
 [1] merge(a::NamedTuple{(), Tuple{}}, itr::Dict{String, Int64})
   @ Base ./namedtuple.jl:293
 [2] top-level scope
   @ REPL[49]:1
```

But we can just convert them (recursively): 
```Julia
_makesymbol(x) = x   # default 
_makesymbol(p::Pair) = (Symbol(p.first) => _makesymbol(p.second))  # recursion 
_makesymbol(D::Dict) = Dict(_makesymbol.([D...])...)    # special case dictionary 
```
e.g. 
```Julia
julia> _makesymbol(Dict("a" => 1, "b" => 2))
Dict{Symbol, Int64} with 2 entries:
  :a => 1
  :b => 2
```
and another example with nested dictionaries 
```Julia
julia> _makesymbol(Dict("a" => 1, "b" => 2, "c" => Dict("d" => 3)))
Dict{Symbol, Any} with 3 entries:
  :a => 1
  :b => 2
  :c => Dict(:d=>3)
```

And now we can splat the kwargs: 
```Julia
julia> D = Dict("a" => 1, "b" => 2); f(; _makesymbol(D)...)
kwargs = Base.Pairs(:a => 1, :b => 2)
pairs(::NamedTuple) with 2 entries:
  :a => 1
  :b => 2
```

Reply:
As long as the Dict that the user passes can be passed into a single function call which then sorts out these parameters, this might "just work", but needs to be carefully tested of course.

Reply:
I wonder though, should we move this to the next PR, so we can wrap this up and tag it?

Reply:
Thanks a lot for the suggestion!! Yes, I would rather not delay too much - have just opened issue #4 and moved your suggestion there.